### PR TITLE
Adding anonymous/default user

### PR DESCRIPTION
**Pull request by VanDavv**
I handled this issue this way - added PreAuthFilter, which searches for the header, an if there isn't any, it gives anon user header. 

Thanks to this solution, we have perfectly hermetic mechanism, which we can very easily turn off.  
However, we need to have anon user in our database, because Authenticator still needs to compare credentials to existing user

### CODE REVIEW

**Comment by mlotysz** [comment]
This is too low level solution for me. First solution was simpler.
Besides, your solution has no code to assure anon user existence.

File: backend/app/src/main/java/com/bls/patronage/auth/BasicAuthenticator.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ private static boolean isMatched(final String plaintextPassword, final String ha
         String email = basicCredentials.getUsername();
         String plaintextPassword = basicCredentials.getPassword();
 
-        final Optional<User> user = Optional.of(userDao.getUserByEmail(email));
+        final Optional<User> user = Optional.fromNullable(userDao.getUserByEmail(email));
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/auth/BasicAuthenticator.java]: But it can't be null.

**Comment by VanDavv** [comment]
First solution was not working, when I passed a request without header, preAuthentication processes in Dropwizard automaticly returned 401, so I had to go before them. New commit removes previously mentioned requirement for the user to exist

**Comment by mlotysz** [comment]
Thanks!

**Pull request merged**